var/spack/repos/builtin/packages/py-biom-format/package.py

```diff
@@ -14,10 +14,12 @@ class PyBiomFormat(PythonPackage):
     homepage = "https://pypi.python.org/pypi/biom-format/2.1.6"
     url      = "https://pypi.io/packages/source/b/biom-format/biom-format-2.1.6.tar.gz"
 
+    version('2.1.7', sha256='b47e54282ef13cddffdb00aea9183a87175a2372c91a915259086a3f444c42f4')
     version('2.1.6', sha256='8eefc275a85cc937f6d6f408d91b7b45eae854cd5d1cbda411a3af51f5b49b0d')
 
     variant('h5py', default=True, description='For use with BIOM 2.0+ files')
 
+    depends_on('python@:3.6', when='@:2.1.7')
     depends_on('py-setuptools', type=('build', 'run'))
     depends_on('py-cython', type='build')
     depends_on('py-h5py', type=('build', 'run'), when='+h5py')
```

Comment:
It looks like this dependency is actually optional. You could add a variant that defaults to False if you want to decrease the number of dependencies. Also, it should be `@2.2.0:`